File: Core kernel Yaiwes/Backend watchdog workflow adaptativo/sandbox-runtime/iii-sandbox/packages/sdk-rust/src/monitor.rs
use std::sync::Arc;

use crate::client::HttpClient;
use crate::error::Result;
use crate::types::{AlertDeleteResponse, AlertHistoryResponse, AlertListResponse, ResourceAlert};

pub struct MonitorManager {
    client: Arc<HttpClient>,
    sandbox_id: String,
}

impl MonitorManager {
    pub fn new(client: Arc<HttpClient>, sandbox_id: String) -> Self {
        Self { client, sandbox_id }
    }

    pub async fn set_alert(
        &self,
        metric: &str,
        threshold: f64,
        action: Option<&str>,
    ) -> Result<ResourceAlert> {
        self.client
            .post(
                &format!(
                    "/sandbox/sandboxes/{}/alerts",
                    self.sandbox_id
                ),
                Some(&serde_json::json!({
                    "metric": metric,
                    "threshold": threshold,
                    "action": action,
                })),
            )
            .await
    }

    pub async fn list_alerts(&self) -> Result<AlertListResponse> {
        self.client
            .get(&format!(
                "/sandbox/sandboxes/{}/alerts",
                self.sandbox_id
            ))
            .await
    }

    pub async fn delete_alert(&self, alert_id: &str) -> Result<AlertDeleteResponse> {
        self.client
            .delete(&format!("/sandbox/alerts/{alert_id}"))
            .await
    }

    pub async fn history(&self, limit: Option<i64>) -> Result<AlertHistoryResponse> {
        let query = match limit {
            Some(l) => format!("?limit={l}"),
            None => String::new(),
        };
        self.client
            .get(&format!(
                "/sandbox/sandboxes/{}/alerts/history{}",
                self.sandbox_id, query
            ))
            .await
    }
}

File: Agente Yaiwes principal/_work/backend-watchdog-full-repair-02/src/iii-sandbox/packages/sdk-python/src/iii_sandbox/monitor.py
from __future__ import annotations

from .client import HttpClient
from .types import ResourceAlert


class MonitorManager:
    def __init__(self, client: HttpClient, sandbox_id: str):
        self._client = client
        self._sandbox_id = sandbox_id

    async def set_alert(
        self,
        metric: str,
        threshold: float,
        action: str | None = None,
    ) -> ResourceAlert:
        body: dict = {"metric": metric, "threshold": threshold}
        if action is not None:
            body["action"] = action
        data = await self._client.post(
            f"/sandbox/sandboxes/{self._sandbox_id}/alerts", body
        )
        return ResourceAlert(**data)

    async def list_alerts(self) -> dict:
        return await self._client.get(
            f"/sandbox/sandboxes/{self._sandbox_id}/alerts"
        )

    async def delete_alert(self, alert_id: str) -> dict:
        return await self._client.delete(f"/sandbox/alerts/{alert_id}")

    async def history(self, limit: int | None = None) -> dict:
        qs = f"?limit={limit}" if limit is not None else ""
        return await self._client.get(
            f"/sandbox/sandboxes/{self._sandbox_id}/alerts/history{qs}"
        )
